game/tests.py

```
@@ -653,22 +781,73 @@ def tearDown(self):
-        r = self.client.get(url)
+        r = self.client.post('/api/check-promotion/', data=json.dumps({'game_state': self.state, 'from_row': 1, 'from_col': 0, 'to_row': 2}), content_type='application/json')
+        self.assertFalse(r.json()['is_promotion'])
+        self.mock_promo.assert_called_once()
+
+class GameStateTest(TestCase):
+    """Test the /api/state/ endpoint."""
+
+    def setUp(self):
+        self.client.get('/play/')
+
+    def _set_game_session(self, game):
```

Ruff (0.15.14)

[warning] Missing return type annotation for private function `_set_game_session`

Add return type annotation: `None`

(ANN202)